include/private/element.h

```diff
@@ -37,9 +53,49 @@ for( i = 0; i < ( ELEMENT )->param_count; i++ ) {   \
     continue;                                       \
   }
 
+/**
+ * Locks the provided element.
+ *
+ * **Thread Safety: MT-Safe**
+ * This function acquires the element's internal lock. Only one thread may hold
+ * the same element's lock at a time.
+ *
+ * **Async Signal Safety: AS-Unsafe lock**
+ * This function is not safe to call from signal handlers due to the use of a
+ * mutex routine.
+ *
+ * **Async Cancel Safety: AC-Unsafe lock**
+ * This function is not safe to call from threads that may be asynchronously
+ * cancelled during the lock operation.
+ *
+ * @param element The element to lock. Must not be NULL.
+ * 
+ */
 void
 lock_element( const struct stumpless_element *element );
 
+/**
+ * Gets a parameter by index from a locked element, without performing additional locking.
+ *
+ * **Thread Safety: MT-Safe req:element-locked**
+ * This function is safe when the caller already holds the element's lock. It
+ * must not be called without the element being locked by the caller.
+ *
+ * **Async Signal Safety: AS-Unsafe**
+ * This function is not intended for use in signal handlers.
```

Comment:
Provide a reason that this is AS-Unsafe, particularly since it is marked as AC-Safe.